components/slack/slack.app.js

```diff
@@ -12,13 +12,14 @@ module.exports = {
         let {
           types,
           cursor,
+          userNames,
         } = prevContext;
         if (types == null) {
-          const scopes = await this.scopes();
           types = [
             "public_channel",
           ];
         }
+        userNames = {};
```

Comment:
Won't this overwrite any `userNames` previously passed from `prevContext` in all cases?

Reply:
Copied the way conversations establishes usernames to public and private channels, should be fixed now :)